### ISSUE TITLE
Colossus: Doesn't gracefully handle committing file to IPFS when IPFS node is down

### ISSUE THREAD

**Issue by mnaamani** [Joystream/joystream#2306]
Colossus exits if tries to commit a new upload when no IPFS daemon is running. 

Simplest fix catch the error, return `5xx` http response code, uploader will need to re-attempt upload.

Alternate fix, return 5xx http response code, but keep the temporary file in cache, re-try to commit to ipfs after some delay. Uploader will not need to re-try but instead wait to see asset 'Accepted' state.

First approach fits better with current Atlas UI/UX design.

```
  joystream:storage:storage Error committing stream Error: connect ECONNREFUSED 127.0.0.1:5001
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16) {
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 5001
} +8ms
  joystream:colossus:api:asset Error: connect ECONNREFUSED 127.0.0.1:5001
  joystream:colossus:api:asset     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1141:16) +0ms
RangeError [ERR_HTTP_INVALID_STATUS_CODE]: Invalid status code: ECONNREFUSED
    at ServerResponse.writeHead (_http_server.js:248:11)
    at ServerResponse._implicitHeader (_http_server.js:239:8)
    at write_ (_http_outgoing.js:642:9)
    at ServerResponse.end (_http_outgoing.js:753:5)
    at ServerResponse.send (/Users/mokhtar/joystream/joystream/node_modules/express/lib/response.js:221:10)
    at ServerResponse.expressOpenAPISend [as send] (/Users/mokhtar/joystream/joystream/storage-node/packages/colossus/lib/middleware/validate_responses.js:34:21)
    at ServerResponse.json (/Users/mokhtar/joystream/joystream/node_modules/express/lib/response.js:267:15)
    at ServerResponse.send (/Users/mokhtar/joystream/joystream/node_modules/express/lib/response.js:158:21)
    at ServerResponse.expressOpenAPISend [as send] (/Users/mokhtar/joystream/joystream/storage-node/packages/colossus/lib/middleware/validate_responses.js:54:21)
    at errorHandler (/Users/mokhtar/joystream/joystream/storage-node/packages/colossus/paths/asset/v0/{id}.js:28:44)
    at StorageWriteStream.<anonymous> (/Users/mokhtar/joystream/joystream/storage-node/packages/colossus/paths/asset/v0/{id}.js:209:37)
    at StorageWriteStream.emit (events.js:315:20)
    at /Users/mokhtar/joystream/joystream/storage-node/packages/storage/storage.js:177:14
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by mnaamani** [comment]
Resolved in #2476

**Comment by shamil-gadelshin** [comment]
Obsolete
